### PR TITLE
Add AutoModel selector in transformers tool

### PR DESCRIPTION
**Pull request by wangyems**
**Description**: Describe your changes.

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: onnxruntime/python/tools/transformers/huggingface_models.py [hunk 1]
```diff
@@ -57,7 +59,7 @@
     "roberta-large-openai-detector": (["input_ids", "attention_mask"], 11, False, "bert"),
 # DistilBERT
     "distilbert-base-uncased": (["input_ids", "attention_mask"], 11, False, "bert"),
-    "distilbert-base-uncased-distilled-squad": (["input_ids", "attention_mask"], 11, False, "bert"), 
+    "distilbert-base-uncased-distilled-squad": (["input_ids", "attention_mask"], 11, False, "bert"),
```
Comment (wangyems) [inline onnxruntime/python/tools/transformers/huggingface_models.py:62]: AutoModelForQuestionAnswering for this?

Comment (tianleiwu) [reply onnxruntime/python/tools/transformers/huggingface_models.py:62]: yes.

**Review by tianleiwu — COMMENTED** [review]
Another way is to use rule base:
- when model name ends with "-squad", use AutoModelForQuestionAnswering
- when model name ends with "-mprc", use AutoModelForSequenceClassification
- when model name has substring "gpt2", use AutoModelWithLMHead
- otherwise, use AutoModel

Shall we allow user to overwrite the setting by parameter like --model_class (as in benchmark_gpt2)?

**Review by tianleiwu — DISMISSED** [review]

File: onnxruntime/python/tools/transformers/onnx_exporter.py [hunk 2]
```diff
@@ -182,20 +183,40 @@ def optimize_onnx_model(onnx_model_path, optimized_model_path, model_type, num_a
...
+
+def load_pretrained_model(model_name, config, cache_dir, pipeline):
+    return pipeline.from_pretrained(model_name, config=config, cache_dir=cache_dir)
+
 
+def export_onnx_model(model_name, model_attribute, model_class, cache_dir, onnx_dir, input_names, use_gpu, precision,
+                      optimize_onnx, validate_onnx, use_raw_attention_mask, overwrite, model_fusion_statistics):
+    opset_version, use_external_data_format, model_type = model_attribute[1], model_attribute[2], model_attribute[3]
 
-def export_onnx_model(model_name, opset_version, use_external_data_format, model_type, cache_dir, onnx_dir, input_names,
```
Comment (tianleiwu) [inline onnxruntime/python/tools/transformers/onnx_exporter.py:191]: It is better to keep old interface that is more straightforward. Consider that this API can be used by others, it is not good to pass model_attribute that might be changed (like add more attribute).

File: onnxruntime/python/tools/transformers/onnx_exporter.py [hunk 3]
```diff
@@ -182,20 +183,40 @@ def optimize_onnx_model(onnx_model_path, optimized_model_path, model_type, num_a
...
-                      use_gpu, precision, optimize_onnx, validate_onnx, use_raw_attention_mask, overwrite,
-                      model_fusion_statistics):
     config = AutoConfig.from_pretrained(model_name, cache_dir=cache_dir)
     if hasattr(config, 'return_dict'):
         config.return_dict = False
 
-    model = load_pretrained_model(model_name, config=config, cache_dir=cache_dir)
+    model = load_pretrained_model(model_name, config=config, cache_dir=cache_dir, pipeline=pipeline_dispatcher(model_name, model_class))
     model.cpu()
```
Comment (tianleiwu) [inline onnxruntime/python/tools/transformers/onnx_exporter.py:220]: load_pretrained_model might return None when model_class does not fit

**Comment by tianleiwu** [comment]
LGTM

**Review by tianleiwu — APPROVED** [review]

**Pull request merged**